sdks/java/core/src/main/java/org/apache/beam/sdk/util/NoopLock.java

```diff
@@ -56,16 +54,15 @@ public boolean tryLock() {
   }
 
   @Override
-  public boolean tryLock(long time, @Nonnull TimeUnit unit) {
+  public boolean tryLock(long time, @NonNull TimeUnit unit) {
     return true;
   }
 
   @Override
   public void unlock() {}
 
-  @Nonnull
   @Override
-  public Condition newCondition() {
+  public @NonNull Condition newCondition() {
```

Comment:
Does this annotation override the superclass? That is safe to do here because it is in a covariant position. I am asking only because non-null is default so it would be unnecessary unless the superclass already specified it was nullable.

Reply:
To answer your question, no. The annotation does not exist on the [Lock](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/locks/Lock.html#newCondition()) interface. The return type was annotated with `javax.annotation.Nonnull` by the original author. Should we leave the `javax.annotation.Nonnull` annotated by the original author or replace with `org.checkerframework.checker.nullness.qual.NonNull`? Thank you for taking the time to look at this.

Reply:
Delete the annotation if it still type checks without it.